invenio_curations/assets/semantic-ui/js/invenio_curations/deposit/withCurationStatus.js

```diff
@@ -0,0 +1,86 @@
+// This file is part of InvenioRDM
+// Copyright (C) 2025 Graz University of Technology.
+//
+// Invenio-Curations is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React from "react";
+import { i18next } from "@translations/invenio_curations/i18next";
+
+/**
+ * HOC that adds curation status information to a component
+ * @param {React.Component} WrappedComponent 
+ * @returns {React.Component}
+ */
+export const withCurationStatus = (WrappedComponent) => {
```

Comment:
should that be moved to a package wide redux store? is it good practice to add this here?

Reply:
No, as this issue is more ui specific and is coupled to the component and using redux store i find it a bit unnecessary for such complexity for what is essentially an ui concern.